codalab/worker/docker_utils.py

```diff
@@ -252,3 +254,17 @@ def check_finished(container):
             failure_msg = 'Memory limit exceeded.'
         return (True, exitcode, failure_msg)
     return (False, None, None)
+
+
+@wrap_exception('Unable to check Docker container running time')
+def get_container_running_time(container):
+    # Identical to the docker inspect container command.
+    container_stats = apiclient.inspect_container(container.id)
+    start_time = container_stats['State']['StartedAt']
+    end_time = (
+        container_stats['State']['FinishedAt']
+        if container_stats['State']['Status'] == 'exited'
+        else datetime.datetime.now().replace(tzinfo=tz.tzutc())
+    )
+    container_running_time = parser.isoparse(end_time) - parser.isoparse(start_time)
+    return container_running_time.seconds
```

Comment:
Instead of creating a raw API client and calling `inspect_container`, you can call `container.reload()` then use the `container.attrs` dict to access the inspection fields of the container. For this purpose the following should work:
```
container.reload()
start_time = container.attrs['State']['StartedAt']
. . .
```
https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container